demo-playground/BlockGeneration.hs

```diff
@@ -22,31 +24,38 @@ forkCoreNode :: forall m p c.
                 ( Mock.SimpleBlockCrypto c
                 , MonadSTM m
                 , MonadTimer m
-                , MonadFork m
                 , MonadIO m
-                , RunOuroboros p (DemoLedgerState (Mock.SimpleBlock p c))
+                , ProtocolLedgerView (Mock.SimpleBlock p c)
                 )
              => TVar m (DemoLedgerState (Mock.SimpleBlock p c))
-             -> OuroborosState p
+             -> OuroborosNodeConfig p
+             -> OuroborosNodeState p
              -> TVar m (Mempool Mock.Tx)
              -> TVar m (ChainProducerState (Mock.SimpleBlock p c))
              -> Duration (Time m)
              -> m ()
-forkCoreNode ledgerVar protocolState mempoolVar varCPS slotDuration = do
+forkCoreNode ledgerVar cfg initState mempoolVar varCPS slotDuration = do
```

Comment:
This feels now quite correct -- after all, it's totally conceivable that a node takes a static config and an initial state, and both are now arguments. I like it! 😉